tool/src/org/antlr/v4/misc/Graph.java

```diff
@@ -44,13 +45,13 @@
 public class Graph<T> {
 
 	public static class Node<T> {
-		T payload;
-		List<Node<T>> edges; // points at which nodes?
+		public T payload;
+		public List<Node<T>> edges = Collections.emptyList(); // points at which nodes?
 
 		public Node(T payload) { this.payload = payload; }
 
 		public void addEdge(Node<T> n) {
-			if ( edges==null ) edges = new ArrayList<Node<T>>();
+			if ( edges==Collections.EMPTY_LIST ) edges = new ArrayList<Node<T>>();
```

Comment:
Thanks for your remark. There is indeed no official contract. emptyList() of Oracle Java and OpenJDK  return EMPTY_LIST and I think other implementations do the same. I use this idiom in a few places and never had any fallout. But I can change it to play safe.